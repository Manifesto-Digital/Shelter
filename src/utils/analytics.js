export const sendEvent = data => {
  window.dataLayer.push(data);
};
