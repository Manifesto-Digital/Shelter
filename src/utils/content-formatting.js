export const addMarksToContent = content => {
  const markup = content
    .map(block => {
      const { marks, value } = block
      // As there may be multiple marks per piece of content, we need to
      // store the opening and closing tags and then wrap the content
      if (marks && marks.length > 0) {
        let openingMarks = ''
        let closingMarks = ''

        marks.forEach(mark => {
          if (mark.type === 'underline') {
            openingMarks = `<span className="underline">${openingMarks}`
            closingMarks = `${closingMarks}</span>`
          }
          if (mark.type === 'italic') {
            openingMarks = `<em>${openingMarks}`
            closingMarks = `${closingMarks}</em>`
          }
          if (mark.type === 'bold') {
            openingMarks = `<strong>${openingMarks}`
            closingMarks = `${closingMarks}</strong>`
          }
        })
        return `${openingMarks}${value}${closingMarks}`
      }
      return value
    })
    .join('')

  return markup
}
