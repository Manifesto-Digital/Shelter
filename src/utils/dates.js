import dayjs from 'dayjs';

// Format date as string in required format
// Formatting options: https://bit.ly/2R9uVsh
export const dateAsString = (date, format) =>
  dayjs(date)
    .format(format)
    .toString();

export const getWeekDays = () => {
  // Base date is a Monday, this ensures we get the day lsit in order
  const baseDate = new Date(Date.UTC(2019, 0, 7));
  const weekDays = [];
  for (let i = 0; i < 7; i += 1) {
    weekDays.push(baseDate.toLocaleDateString('en-GB', { weekday: 'long' }));
    baseDate.setDate(baseDate.getDate() + 1);
  }
  return weekDays;
};

export const dayOfTheWeek = () => {
  const daysArray = getWeekDays();
  const today = new Date();

  return daysArray[today.getDay() - 1];
};
