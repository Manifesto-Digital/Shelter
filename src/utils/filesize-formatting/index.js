/**
 * Formats integer to human readable filesize
 *
 * @param {int} bytes
 * @param {int} decimalPoint
 */
export const formatFilesize = (bytes, decimalPoint) => {
  if (bytes === undefined) throw new Error('Bytes are required');
  if (bytes < 0) throw new Error('Negative bytes not allowed');
  if (bytes >= 10 ** 27)
    throw new Error(`Bytes too large, must be < 10 ** 27 (${10 ** 27})`);

  if (bytes === 0) return '0 Bytes';

  const k = 1000;
  const dm = decimalPoint || 2;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  const sizeToReturn = `${parseFloat((bytes / k ** i).toFixed(dm))} ${
    sizes[i]
  }`;
  return sizeToReturn;
};
