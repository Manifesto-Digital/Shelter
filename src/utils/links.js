export const getInternalLink = slug => {
  const baseURL = window.location.origin
  return `${baseURL}/${slug}`
}
