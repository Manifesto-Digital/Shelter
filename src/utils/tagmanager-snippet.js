const tagManagerSnippet = () =>
  (function(w, d, s, l, i) {
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    const f = d.getElementsByTagName(s)[0];
    const j = d.createElement(s);
    const dl = l != 'dataLayer' ? `&l=${l}` : ''; // eslint-disable-line eqeqeq
    j.async = true;
    j.src = `https://www.googletagmanager.com/gtm.js?id=${i}${dl}`;
    f.parentNode.insertBefore(j, f);
  })(window, document, 'script', 'dataLayer', 'GTM-WWSW2HH');

export default tagManagerSnippet;
