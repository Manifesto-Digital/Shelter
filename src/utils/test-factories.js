// We define some factories for creating the various types of objects in our system.
// These are just functions that take `Partial<T>` and return `T`, assigning defaults to
// the properties that aren't specified.

export const createFactory = defaults => properties => ({
  ...defaults,
  ...properties,
});

export const createInternalLink = createFactory({
  id: 'ee6c2ca6-54d8-5c26-bbff-7dafdaa823e3',
  title: 'Test Page Title',
  slug: 'internal-link-slug',
});

export const createImage = createFactory({
  id: 'ca8e4a9c-9491-5257-8e53-d3d74ae204a5',
  description: 'Collection hero',
  title: 'collection-hero',
  file: {
    url:
      '//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg',
    fileName: 'collection-hero.jpg',
    contentType: 'image/jpeg',
  },
});

export const createVideo = createFactory({
  id: 'ca8e4a9c-9491-5257-8e53-d3d74ae204a5',
  title: 'Urban Rush hero',
  description: 'Urban rush video',
  file: {
    url:
      '//videos.ctfassets.net/6sxvmndnpn0s/5hlfsk9dwVVNBlbpWHSLvK/9f1d647e6332be389465c9d014a57a26/shelter-urban-rush.mp4',
    fileName: 'shelter-urban-rush.mp4',
    contentType: 'video/mp4',
  },
});

export const createChildContentfulRichText = createFactory({
  json: {
    data: {},
    content: [
      {
        data: {},
        content: [
          {
            data: {},
            marks: [],
            value: 'Some body text',
            nodeType: 'text',
          },
        ],
        nodeType: 'paragraph',
      },
      {
        data: {},
        content: [
          {
            data: {},
            marks: [],
            value: 'Some header text',
            nodeType: 'text',
          },
        ],
        nodeType: 'heading-2',
      },
      {
        data: {},
        content: [
          {
            data: {},
            marks: [
              {
                type: 'bold',
              },
            ],
            value: 'some bollld text',
            nodeType: 'text',
          },
        ],
        nodeType: 'paragraph',
      },
    ],
  },
  childContentfulRichText: {
    html:
      '<p>Our advice and support services across the UK give people <b>one-to-one</p>',
  },
});

export const createFile = createFactory({
  description: 'Mock file description',
  title: 'Mock file title',
  file: {
    contentType: 'application/pdf',
    url:
      '//assets.ctfassets.net/6sxvmndnpn0s/6gDmPHZ04gCMMS4sOa0IEk/7c712c5e8e383bbafd61f2d9d570de4f/component_tree-pdf.pdf',
    fileName: 'component tree-pdf.pdf',
    details: {
      size: 78019,
      image: null,
    },
  },
});

export const createHeaderNavigation = createFactory({
  id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
  name: 'Assembly - Primary Navigation',
  additionalLink: [
    {
      id: 'e41df6b6-52e2-52db-93c7-9d7ecef7fcd4',
      title: 'Tobys Page',
      slug: 'tobys-page',
    },
  ],
  menuLabel: 'What we do',
  navigationItems: [
    {
      id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
      menuLabel: 'Housing Advice',
      navigationLink: [{ title: 'Tobys Page', slug: 'tobys-page' }],
      subNavigationItems: [
        {
          slug: '320,000-people-in-britain-are-now-homeless',
          title: '320,000 people in Britain are now homeless',
        },
        {
          title: 'Tobys Page',
          slug: 'tobys-page',
        },
        {
          title: 'Shelter Manchester to hold vigil for lost homeless people',
          slug: 'shelter-manchester-to-hold-vigil-for-lost-homeless-people',
        },
        {
          title: 'Shelter Demo Page',
          slug: 'shelter-demo-page',
        },
      ],
    },
  ],
  footerText: createChildContentfulRichText(),
});

export const createSubpage = createFactory({
  id: '27757b6a-db0e-5aca-8db2-8f3e736b5341',
  slug: 'advice-guide-test',
  title: 'Advice guide test',
  subPages: null,
});

export const createContentCards = createFactory({
  pageInformation: {
    pageThumbnail: {
      description: 'Telli Afrik and family',
      file: {
        url:
          '//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg',
        fileName: 'Telli Afrik and family - Steve Franck.jpg',
      },
    },
    shortDescription: {
      shortDescription: 'A short description for Callams test page',
    },
  },
  slug: 'shelter-demo-page',
  title: 'Shelter Demo Page',
});

export const createContentCardBanner = createFactory({
  header: createChildContentfulRichText(),
  bannerColour: 'Grey',
  bannerFlow: 'Horizontal',
  contentCards: [createContentCards(), createContentCards()],
});

export const createTestRef = createFactory({
  current: {},
});

export const createEvent = createFactory({
  eventName: 'Brighton Half Marathon',
  shortDescription:
    'Run for charity with Team Shelter; half marathons, marathons and other distance running events all in the name of a good cause.',
  thumbnailImage: createImage(),
  eventType: 'Run',
  eventStatus: 'Sold Out',
  eventSystemDate: '2019-02-24',
  eventDisplayDate: '24 February 2019',
  distance: '13.1 miles',
  registrationFee: 19,
  pledge: 199,
  eventLocation: {
    lat: 50.8374204,
    lon: -0.1061897,
  },
  displayLocation: 'Brighton',
  notifiedTeamEmail: 'running@shelter.org.uk',
});

export const createPerson = createFactory({
  id: 'd5ec5ab5-6824-5488-b9bc-4d8794fba4c6',
  firstName: 'Toby',
  lastName: 'Bushell',
  phoneNumber: '012345678910',
  emailAddress: 'test@test.com',
  photo: createImage(),
  bio: {
    internal: {
      content:
        'Nunc nonummy metus. Fusce pharetra convallis urna. Praesent metus tellus, elementum eu, semper a, adipiscing nec, purus.',
    },
  },
  jobTitle: 'Developer',
});

export const createInternalRef = createFactory({
  internal: {
    type: 'ContentfulPageAssemblyContentPage',
  },
  slug: 'internal-link-slug',
});

export const createExternalRef = createFactory({
  internal: {
    type: 'ContentfulTopicExternalLink',
  },
  URL: 'http://mock.com',
  newTab: false,
});
