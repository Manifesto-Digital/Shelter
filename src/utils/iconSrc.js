import SpeechBubble from '../assets/svg/icons/comment-light.svg';
import Envelope from '../assets/svg/icons/envelope-light.svg';
import Download from '../assets/svg/icons/download-light.svg';
import Printer from '../assets/svg/icons/print-light.svg';
import OpenBook from '../assets/svg/icons/book-open-light.svg';
import Exclamation from '../assets/svg/icons/exclamation-circle-light.svg';
import Telephone from '../assets/svg/icons/phone-light.svg';
import MapMarker from '../assets/svg/icons/map-marker-alt-light.svg';
import ChippedTimeResults from '../assets/svg/icons/challenge-event/chipped-time-results.svg';
import FinishersMedal from '../assets/svg/icons/challenge-event/finishers-medal.svg';
import FreeBagTransfer from '../assets/svg/icons/challenge-event/free-bag-transfer.svg';
import FreeRacePhotos from '../assets/svg/icons/challenge-event/free-race-photos.svg';
import GoodieBags from '../assets/svg/icons/challenge-event/goodie-bags.svg';
import PostRaceCelebration from '../assets/svg/icons/challenge-event/post-race-celebration.svg';
import SportsMassage from '../assets/svg/icons/challenge-event/sports-massage.svg';
import TechnicalRunningTop from '../assets/svg/icons/challenge-event/technical-running-top.svg';
import ClipboardChecklist from '../assets/svg/icons/clipboard-checklist-light.svg';
import FundingGoalMeter from '../assets/svg/icons/funding-goal-meter-light.svg';
import Lightbulb from '../assets/svg/icons/lightbulb-light.svg';
import MapUnfolded from '../assets/svg/icons/map-unfolded-light.svg';

import { consistentString } from './content-formatting';

const iconList = {
  'speech-bubble': SpeechBubble,
  envelope: Envelope,
  download: Download,
  printer: Printer,
  'open-book': OpenBook,
  'exclamation-mark': Exclamation,
  telephone: Telephone,
  'map-marker': MapMarker,
  'chipped-time-results': ChippedTimeResults,
  'finishers-medal': FinishersMedal,
  'free-bag-transfer': FreeBagTransfer,
  'free-race-photos': FreeRacePhotos,
  'goodie-bags': GoodieBags,
  'post-race-celebration': PostRaceCelebration,
  'sports-massage': SportsMassage,
  'technical-running-top': TechnicalRunningTop,
  'clipboard-checklist': ClipboardChecklist,
  'funding-goal-meter': FundingGoalMeter,
  lightbulb: Lightbulb,
  'map-unfolded': MapUnfolded,
  'no-icon': null,
  default: null,
};

// returns the src of an svg icon from the above icon list
const iconSrc = icon => {
  const iconKey = consistentString(icon);

  if (iconList[iconKey] === undefined) {
    throw new Error(
      'No matching icon was found. Is it set as an option in the CMS? Has it been imported and included in the object above?'
    );
  }

  const chosenIcon = iconList[iconKey] || null;

  if (!chosenIcon) return null;

  return chosenIcon;
};

export default iconSrc;
