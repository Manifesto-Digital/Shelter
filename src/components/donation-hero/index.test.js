import React from 'react';
import Slider from 'rc-slider/lib/Slider';
import { mountWithTheme, snapshotComponent } from 'test-helpers';
import {
  StyledDonateButton,
  CollapsableArea,
  OwnAmountButton,
  OwnAmountSubmit,
} from './styles';
import DonationForm from '../donation-form-handler';
import { createFactory, createImage } from '../../utils/test-factories';
import DonationHero from '.';

// Default props
export const createDonationHero = createFactory({
  heading: 'Your donations make a difference',
  singleAmount1: 10,
  singleDescription1:
    'Just £10 could pay for two web chats providing essential housing advice.',
  singleImage1: createImage(),
  singleAmount2: 20,
  singleDescription2:
    'Donating £20 could answer two urgent calls to our helpline.',
  singleImage2: createImage(),
  singleAmount3: 30,
  singleDescription3:
    'Giving £30 could pay for a face-to-face advice session with a Shelter adviser.',
  singleImage3: createImage(),
  singleAmount4: 50,
  singleDescription4:
    'Giving £50 could answer five urgent calls to our helpline.',
  singleImage4: createImage(),
  singleAmount5: 80,
  singleDescription5:
    'Donating £80 could help pay for legal advice that could help a family keep their home.',
  singleImage5: createImage(),
  monthlyAmount1: 5,
  monthlyDescription1:
    'Just £5 could pay for a web chat providing essential housing advice.',
  monthlyImage1: createImage(),
  monthlyAmount2: 10,
  monthlyDescription2:
    'Donating £10 can answer an urgent call to our helpline.',
  monthlyImage2: createImage(),
  monthlyAmount3: 15,
  monthlyDescription3:
    'Giving £15 could pay for three web chats providing essential housing advice.',
  monthlyImage3: createImage(),
  monthlyAmount4: 20,
  monthlyDescription4:
    'Donating £20 could help pay for a face-to-face advice session with a Shelter adviser.',
  monthlyImage4: createImage(),
});

const expectAmount = (wrapper, { value, description, image }) => {
  expect(wrapper.find('img').props().src).toContain(image.file.url);
  expect(wrapper.find(Slider).prop('value')).toBe(value);
  expect(wrapper.find(StyledDonateButton).text()).toContain(`Donate £${value}`);
  expect(
    wrapper.find('#donation-hero-form input[name="amount"]').props().value
  ).toEqual((value * 100).toString()); // Amount used in form submission (pence)
  expect(wrapper.text()).toContain(description);
};

it('renders correctly', () => {
  const mockData = createDonationHero();
  snapshotComponent(<DonationHero content={mockData} />);
});

it('initially displays 3rd single payment option', () => {
  const mockData = createDonationHero({
    singleAmount3: 35,
    singleDescription3: 'Mock singleDescription3 description',
    singleImage3: createImage({
      description: 'Collection hero',
      file: {
        url:
          '//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg',
      },
    }),
  });

  const wrapper = mountWithTheme(<DonationHero content={mockData} />);

  expectAmount(wrapper, {
    value: mockData.singleAmount3,
    description: mockData.singleDescription3,
    image: mockData.singleImage3,
  });
});

it('switches option values when changing slider', () => {
  const mockData = createDonationHero();
  const wrapper = mountWithTheme(<DonationHero content={mockData} />);

  // Defaults
  expectAmount(wrapper, {
    value: mockData.singleAmount3,
    description: mockData.singleDescription3,
    image: mockData.singleImage3,
  });

  const onChange = wrapper.find(Slider).prop('onChange');
  onChange(mockData.singleAmount5);

  wrapper.update();

  expectAmount(wrapper, {
    value: mockData.singleAmount5,
    description: mockData.singleDescription5,
    image: mockData.singleImage5,
  });
});

it('renders a donation form for the slider and a donation form for the other amount', () => {
  const mockData = createDonationHero();
  const wrapper = mountWithTheme(<DonationHero content={mockData} />);
  expect(wrapper.find(DonationForm)).toHaveLength(2);
});

it('starts with donation form collapsed', () => {
  const mockData = createDonationHero();
  const wrapper = mountWithTheme(<DonationHero content={mockData} />);
  expect(wrapper.find(CollapsableArea)).toHaveStyleRule('max-height', '0');
});

it('expands other amount donation form when clicking choosing to enter own amount', () => {
  const mockData = createDonationHero();
  const wrapper = mountWithTheme(<DonationHero content={mockData} />);

  wrapper.find(OwnAmountButton).simulate('click');

  expect(wrapper.find(CollapsableArea)).not.toHaveStyleRule('max-height', '0');
});

it('both forms have a button to submit', () => {
  const mockData = createDonationHero();
  const wrapper = mountWithTheme(<DonationHero content={mockData} />);

  expect(wrapper.find(StyledDonateButton)).toHaveLength(1);
  expect(wrapper.find(OwnAmountSubmit)).toHaveLength(1);
});
