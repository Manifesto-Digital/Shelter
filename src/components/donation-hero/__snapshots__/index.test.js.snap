// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c12 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #12b9bf;
  border: none;
  color: #fff;
  width: 100%;
}

.c20 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #12b9bf;
  border: none;
  color: #fff;
}

.c0 {
  position: relative;
  overflow: hidden;
  margin-bottom: 40px;
}

.c1 {
  max-height: 200px;
}

.c4 {
  background: #d9d9d9;
  position: relative;
}

.c5 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 > li {
  margin: 0;
}

.c6 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  padding: 20px;
  text-align: center;
}

.c6:not(.is-selected) {
  background: #f7f7f7;
  color: #999;
}

.c6:hover {
  color: #ea232d;
}

.c7 {
  padding: 20px 20px 10px 20px;
}

.c11 {
  margin-bottom: 20px;
}

.c8 {
  margin-bottom: 20px;
}

.c14 {
  max-height: 0;
  -webkit-transition: max-height 0.2s ease-in-out;
  transition: max-height 0.2s ease-in-out;
  overflow: hidden;
}

.c3 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
  padding: 0;
}

.c13 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  color: #4d63b5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 700;
  display: block;
  margin-bottom: 20px;
}

.c13:hover {
  color: #ea232d;
}

.c19 {
  margin-left: 10px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  position: relative;
  padding-top: 45px;
  margin-left: 10px;
  margin-right: 10px;
  margin-bottom: calc( 40px + 10px );
}

.c16 {
  position: relative;
}

.c18 {
  display: inline-block;
  background-size: 15px 15px;
  width: auto;
  margin-bottom: 0;
  padding: 10px;
  padding-left: 30px;
  max-width: 10em;
}

.c18::-webkit-outer-spin-button,
.c18::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c17 {
  left: 12px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  z-index: 1;
  font-size: 0.775rem;
}

.c9 {
  padding: 0;
  margin: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(1px);
  clip-path: inset(1px);
  width: 1px;
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

@media (min-width:62em) {
  .c0 {
    height: 660px;
  }
}

@media (min-width:40.0625em) {
  .c1 {
    max-height: 300px;
  }
}

@media (min-width:62em) {
  .c1 {
    max-height: none;
  }
}

@media (min-width:62em) {
  .c2 {
    position: absolute;
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    top: 0;
    margin: auto;
    width: 100%;
  }
}

@media (min-width:62em) {
  .c4 {
    width: 55%;
    margin-top: 60px;
  }
}

@media (min-width:40.0625em) {
  .c7 {
    padding: 40px 40px 20px 40px;
  }
}

@media (min-width:62em) {
  .c8 {
    min-height: 3em;
  }
}

@media (min-width:62em) {
  .c3 {
    padding: 0 20px;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Collection hero"
      className="c2"
      src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fm=jpg&fl=progressive&w=1500"
      srcSet="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fm=jpg&fl=progressive&w=700 480w,
        //images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?fm=jpg&fl=progressive&w=1500"
    />
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <ul
          className="c5"
          role="tablist"
        >
          <li
            aria-controls="react-tabs-1"
            aria-disabled="false"
            aria-selected="true"
            className="c6 is-selected"
            id="react-tabs-0"
            role="tab"
            tabIndex="0"
          >
            Give once
          </li>
          <li
            aria-controls="react-tabs-3"
            aria-disabled="false"
            aria-selected="false"
            className="c6"
            id="react-tabs-2"
            role="tab"
            tabIndex={null}
          >
            Give monthly
          </li>
        </ul>
        <div
          aria-labelledby="react-tabs-0"
          className="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="react-tabs-1"
          role="tabpanel"
        >
          <section
            className="c7"
          >
            <h2>
              Your donations make a difference
            </h2>
            <p
              className="c8"
            >
              Giving Â£30 could pay for a face-to-face advice session with a Shelter adviser.
            </p>
            <form
              action="https://donate.shelter.org.uk/b"
              className=""
              id="donation-hero-form"
              inline={0}
              method="GET"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <input
                name="cid"
                onBlur={[Function]}
                onChange={[Function]}
                type="hidden"
                value="263"
              />
              <input
                name="free_amount"
                onBlur={[Function]}
                onChange={[Function]}
                type="hidden"
                value="1"
              />
              <input
                name="amount"
                onBlur={[Function]}
                onChange={[Function]}
                type="hidden"
                value="3000"
              />
              <input
                name="reserved_appeal_code"
                onBlur={[Function]}
                onChange={[Function]}
                type="hidden"
                value=""
              />
              <input
                name="frequency"
                onBlur={[Function]}
                onChange={[Function]}
                type="hidden"
                value="once"
              />
              <label
                className="c9"
                htmlFor="amount-holder"
              >
                Donate
              </label>
              <div
                className="rc-slider rc-slider-with-marks c10"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  className="rc-slider-rail"
                  style={
                    Object {
                      "backgroundColor": "#bdbec0",
                      "borderRadius": 5,
                      "height": 10,
                      "position": "absolute",
                      "width": "100%",
                    }
                  }
                />
                <div
                  className="rc-slider-track"
                  style={
                    Object {
                      "backgroundColor": "#5a5a5a",
                      "borderRadius": 5,
                      "height": 10,
                      "left": "0%",
                      "position": "absolute",
                      "width": "28.57142857142857%",
                    }
                  }
                />
                <div
                  className="rc-slider-step"
                >
                  <span
                    className="rc-slider-dot rc-slider-dot-active"
                    style={
                      Object {
                        "left": "0%",
                      }
                    }
                  />
                  <span
                    className="rc-slider-dot rc-slider-dot-active"
                    style={
                      Object {
                        "left": "14.285714285714285%",
                      }
                    }
                  />
                  <span
                    className="rc-slider-dot rc-slider-dot-active"
                    style={
                      Object {
                        "left": "28.57142857142857%",
                      }
                    }
                  />
                  <span
                    className="rc-slider-dot"
                    style={
                      Object {
                        "left": "57.14285714285714%",
                      }
                    }
                  />
                  <span
                    className="rc-slider-dot"
                    style={
                      Object {
                        "left": "100%",
                      }
                    }
                  />
                </div>
                <div
                  aria-disabled={false}
                  aria-valuemax={80}
                  aria-valuemin={10}
                  aria-valuenow={30}
                  className="rc-slider-handle"
                  onBlur={[Function]}
                  onKeyDown={[Function]}
                  role="slider"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "border": "1px solid #d9d9d9",
                      "borderRadius": "50%",
                      "boxShadow": "0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06)",
                      "height": 30,
                      "left": "28.57142857142857%",
                      "marginLeft": -15,
                      "marginTop": -10,
                      "position": "absolute",
                      "width": 30,
                    }
                  }
                  tabIndex={0}
                />
                <div
                  className="rc-slider-mark"
                >
                  <span
                    className="rc-slider-mark-text rc-slider-mark-text-active"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                        "left": "0%",
                        "lineHeight": "35px",
                        "marginLeft": "-11.25%",
                        "position": "absolute",
                        "textAlign": "center",
                        "top": 0,
                        "verticalAlign": "middle",
                        "width": "22.5%",
                      }
                    }
                  >
                    Â£10
                  </span>
                  <span
                    className="rc-slider-mark-text rc-slider-mark-text-active"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                        "left": "14.285714285714285%",
                        "lineHeight": "35px",
                        "marginLeft": "-11.25%",
                        "position": "absolute",
                        "textAlign": "center",
                        "top": 0,
                        "verticalAlign": "middle",
                        "width": "22.5%",
                      }
                    }
                  >
                    Â£20
                  </span>
                  <span
                    className="rc-slider-mark-text rc-slider-mark-text-active"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontSize": "1.25em",
                        "fontWeight": "bold",
                        "left": "28.57142857142857%",
                        "lineHeight": "35px",
                        "marginLeft": "-11.25%",
                        "position": "absolute",
                        "textAlign": "center",
                        "top": 0,
                        "verticalAlign": "middle",
                        "width": "22.5%",
                      }
                    }
                  >
                    Â£30
                  </span>
                  <span
                    className="rc-slider-mark-text"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                        "left": "57.14285714285714%",
                        "lineHeight": "35px",
                        "marginLeft": "-11.25%",
                        "position": "absolute",
                        "textAlign": "center",
                        "top": 0,
                        "verticalAlign": "middle",
                        "width": "22.5%",
                      }
                    }
                  >
                    Â£50
                  </span>
                  <span
                    className="rc-slider-mark-text"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                        "left": "100%",
                        "lineHeight": "35px",
                        "marginLeft": "-11.25%",
                        "position": "absolute",
                        "textAlign": "center",
                        "top": 0,
                        "verticalAlign": "middle",
                        "width": "22.5%",
                      }
                    }
                  >
                    Â£80
                  </span>
                </div>
              </div>
              <button
                className="c11 c12"
                type="submit"
              >
                Donate Â£
                30
              </button>
            </form>
            <button
              className="c13"
              onClick={[Function]}
            >
              Choose your own amount
            </button>
            <div
              className="c14"
            >
              <form
                action="https://donate.shelter.org.uk/b"
                className="c15"
                id="donation-hero-own-amount-form"
                inline={1}
                method="GET"
                onReset={[Function]}
                onSubmit={[Function]}
              >
                <input
                  name="cid"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="263"
                />
                <input
                  name="free_amount"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="1"
                />
                <input
                  name="amount"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="3000"
                />
                <input
                  name="reserved_appeal_code"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value=""
                />
                <input
                  name="frequency"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="hidden"
                  value="once"
                />
                <label
                  className="c9"
                  htmlFor="amount-holder-other-amount"
                >
                  Donate
                </label>
                <div
                  className="c16"
                >
                  <span
                    className="c17"
                  >
                    Â£
                  </span>
                  <input
                    className="c18"
                    id="amount-holder-other-amount"
                    name="amount-holder"
                    onBlur={[Function]}
                    onChange={[Function]}
                    pattern="[0-9]*"
                    placeholder="30"
                    type="number"
                    value=""
                  />
                </div>
                <button
                  className="c19 c20"
                  type="submit"
                >
                  Donate
                </button>
              </form>
            </div>
          </section>
        </div>
        <div
          aria-labelledby="react-tabs-2"
          className="react-tabs__tab-panel"
          id="react-tabs-3"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</section>
`;
